nexus/reconfigurator/planning/src/blueprint_builder/builder.rs

```diff
@@ -434,15 +435,22 @@ impl<'a> BlueprintBuilder<'a> {
     ) -> Result<&mut BuilderExternalNetworking<'a>, Error> {
         self.external_networking
             .get_or_try_init(|| {
-                BuilderExternalNetworking::new(
+                // Check the planning input: there shouldn't be any external
+                // networking resources in the database (the source of `input`)
+                // that we don't know about from the parent blueprint.
+                ensure_input_networking_records_appear_in_parent_blueprint(
```

Comment:
This was previously called inside `BuilderExternalNetworking::new()`; I moved it out here because it's the only thing that actually looked at the parent blueprint, and moving it made it much simpler to test `BuilderExternalNetworking` in isolation (it no longer requires constructing an entire blueprint).